bin/git-submitpr

```diff
@@ -12,9 +12,19 @@ gitlab_mode_enabled=$(git config --default false --type=bool pile.gitlabModeEnab
 gitlab_url=$(git config --default '' pile.gitlabMergeRequestsURL)
 
 if [[ $gitlab_mode_enabled = true ]]; then
-  if [[ -z $gitlab_url ]]; then
-    echo "error: GitLab is enabled, but no URL is set, run 'git config pile.gitlabMergeRequestsURL \"https://gitlab.com/{group}/{project}/-/merge_requests/new\"' to configure it" >&2
-    exit 1
+  if [[ -z "$gitlab_url" ]]; then
+    origin_url=$(git remote get-url origin | sed 's/\.git$//')
+    ssh_pattern="^git@(git\.)?([^:]+):(.+)$"
+    https_pattern="^https:\/\/(.+)$"
+    if [[ $origin_url =~ $ssh_pattern ]]; then
+      gitlab_url="https://${BASH_REMATCH[2]}/${BASH_REMATCH[3]}/-/merge_requests/new"
+    elif [[ $origin_url =~ $https_pattern ]]; then
+      gitlab_url="https://${BASH_REMATCH[1]}/-/merge_requests/new"
+    else
```

Comment:
>separate PR sounds good. i guess the biggest reason for allowing overriding is if the url doesn't contain gitlab or github at all? because otherwise these look pretty derivable

I thought about unusual git URLs like https://stackoverflow.com/q/31801271, but in fact if we use GitLab it will be `ssh` or `https`, so we can keep it simple and assume that we always can derive the right URL.
I will simplify this code tomorrow.